commands/credentials/destroy.js

```diff
@@ -0,0 +1,47 @@
+'use strict'
+
+const co = require('co')
+const cli = require('heroku-cli-util')
+
+function * run (context, heroku) {
+  const fetcher = require('../../lib/fetcher')(heroku)
+  const host = require('../../lib/host')
+  const util = require('../../lib/util')
+
+  const {app, args, flags} = context
+  let cred = flags.name
+
+  let db = yield fetcher.addon(app, args.database)
+  if (util.starterPlan(db)) throw new Error('This operation is not supported by Hobby tier databases.')
+
+  let attachments = yield heroku.get(`/addons/${db.name}/addon-attachments`)
+  let credAttachments = attachments.filter(a => a.namespace === `credential:${flags.name}`)
+  if (credAttachments.length > 0) throw new Error(`Credential ${flags.name} must be detached from all other apps before destroying.`)
```

Comment:
There actually is already a story for that :) working through a few higher priority items at the moment, but expect a patch in a few days :)